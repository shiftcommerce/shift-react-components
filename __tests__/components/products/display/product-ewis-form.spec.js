// Libraries
import React from 'react'

// Components
import ProductEwisForm from '../../../../src/components/products/display/product-ewis-form'

test('renders correctly', () => {
  // Arrange & Act
  const wrapper = mount(
    <ProductEwisForm />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('input')).toHaveClassName('c-ewis-form__input-field')
  expect(wrapper.find('button')).toIncludeText('Email When in Stock')
})
