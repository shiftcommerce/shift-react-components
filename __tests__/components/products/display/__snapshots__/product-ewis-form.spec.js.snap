// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ProductEwisForm>
  <_class
    className="c-ewis-form__input-field"
    placeholder="Email"
  >
    <Input
      className="c-ewis-form__input-field"
      placeholder="Email"
      renderValidationMessage={[Function]}
    >
      <div
        className="o-form__input-group"
      >
        <label
          className="o-form__label"
        >
          <b>
             
          </b>
          <input
            checked={false}
            className="o-form__input-field c-ewis-form__input-field"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email"
          />
        </label>
      </div>
    </Input>
  </_class>
  <Button
    aria-label="Add to Basket"
    className="c-product-display__buttons-basket o-button--sml"
    label="Email When in Stock"
    onClick={[Function]}
    status="positive"
  >
    <button
      aria-label="Add to Basket"
      className="c-product-display__buttons-basket o-button--sml o-button o-button--positive"
      disabled={false}
      onClick={[Function]}
      role="button"
      tabIndex="0"
    >
      Email When in Stock
    </button>
  </Button>
</ProductEwisForm>
`;
